crates/rust-analyzer/src/reload.rs

```diff
@@ -81,9 +82,10 @@ impl GlobalState {
                 &self.config.lru_query_capacities_config().cloned().unwrap_or_default(),
             );
         }
+
         if self.config.linked_or_discovered_projects() != old_config.linked_or_discovered_projects()
         {
-            self.fetch_workspaces_queue.request_op("discovered projects changed".to_owned(), false)
+            self.fetch_workspaces_queue.request_op("discovered projects changed".to_owned(), true)
```

Comment:
I can change this to an enum, since I'm doing a bunch of surgery here anyways: would something like (bad namesincoming!) `ReloadBehavior { Force, Done }` be a decent starting point?